app/src/main/java/protect/card_locker/LoyaltyCardViewActivity.java

```diff
@@ -146,6 +147,21 @@ public void onMainImageTap() {
         openImageInGallery(imageType);
     }
 
+    private boolean copyBarcodeToClipBoard(){
+        if (imageTypes.get(mainImageIndex) == ImageType.BARCODE) {
+            String barcodeString = barcodeIdString != null ? barcodeIdString : cardIdString;
+            ClipboardManager clipboard = (ClipboardManager) getSystemService(CLIPBOARD_SERVICE);
+            if (clipboard != null) {
+                android.content.ClipData clip = android.content.ClipData.newPlainText("Barcode", barcodeString);
```

Comment:
I think most users want to copy the `cardId`, not the `barcodeId`.

CardID = the value written on the physical card (mostly the customer number)
BarcodeID = whatever is in the barcode itself (which may contain random gibberish)

The cardId is also what's written in the dialog window.

```suggestion
            ClipboardManager clipboard = (ClipboardManager) getSystemService(CLIPBOARD_SERVICE);
            if (clipboard != null) {
                android.content.ClipData clip = android.content.ClipData.newPlainText("Card ID", loyaltyCard.cardId);
```